recipes/Aishell1Mix/meta/preprocess_dynamic_mixing.py

```diff
@@ -0,0 +1,98 @@
+"""
+This script allows to resample a folder which contains audio files.
+The files are parsed recursively. An exact copy of the folder is created,
+with same structure but contained resampled audio files.
+Resampling is performed by using sox through torchaudio.
+Author
+------
+Samuele Cornell, 2020
+"""
+
+import os
+import argparse
+from pathlib import Path
+import tqdm
+import torchaudio
+import glob
+
+# from oct2py import octave
+from scipy import signal
+import numpy as np
+import torch
+
+
+parser = argparse.ArgumentParser(
+    "utility for resampling all audio files in a folder recursively"
+    "It --input_folder to --output_folder and "
+    "resamples all audio files with specified format to --fs."
+)
+parser.add_argument("--input_folder", type=str, required=True)
+parser.add_argument("--output_folder", type=str, required=True)
+parser.add_argument(
+    "--fs", type=str, default=8000, help="this is the target sampling frequency"
+)
+parser.add_argument("--regex", type=str, default="**/*.wav")
+
+
+def resample_folder(input_folder, output_folder, fs, regex):
+    """Resamples the wav files within an input folder.
+
+    Arguments
+    ---------
+    input_folder : path
+        Path of the folder to resample.
+    output_folder : path
+        Path of the output folder with the resampled data.
+    fs : int
+        Target sampling frequency.
+    reg_exp: str
+        Regular expression for search.
+    """
+    # filedir = os.path.dirname(os.path.realpath(__file__))
+    # octave.addpath(filedir)
+    # add the matlab functions to octave dir here
+
+    files = glob.glob(os.path.join(input_folder, regex), recursive=True)
+    for f in tqdm.tqdm(files):
+
+        audio, fs_read = torchaudio.load(f)
+        audio = audio[0].numpy()
+        audio = signal.resample_poly(audio, fs, fs_read)
+
+        if len(audio) == 0:
```

Comment:
Hi @ycemsubakan , aishell1 contains such zero length audio which cause this error so I insert this line to skip. But the aishell1mix generation has filtered out length < 3 audio so don't worry.